### ISSUE TITLE
Haven't found where command replies

### ISSUE THREAD

**Issue by akmitrich** [KaranGauswami/freeswitch-esl#13]
I am to create an ESL server to control FreeSWITCH calls. I try to follow an outbound socket example. So my FreeSWITCH sent an event stream to my ESL server and ESL server issued a playback command on FreeSWITCH but after this the server got stuck. Logs revealed that the server never returns from [playback call](https://github.com/KaranGauswami/freeswitch-esl/blob/df6dc403c15e55ea93f03ca4f8b3a579019180e4/examples/outbound.rs#L7C5-L7C49). I started reading the source and found there the method [send_recv](https://github.com/KaranGauswami/freeswitch-esl/blob/df6dc403c15e55ea93f03ca4f8b3a579019180e4/src/connection.rs#L69) of the `EslConnection`. The method awaits on a oneshot channel which is ok but I haven't found anywhere in the source code where any reply is sent via this channel.
Please tell me, am I doing something wrong? or is this a bug? or is this a work in progress?

**Comment by KaranGauswami** [comment]
Hi @akmitrich ,

Could you please share a minimal example so that I can review and check it?

**Comment by akmitrich** [comment]
The example is quite close to [outbound.rs](https://github.com/KaranGauswami/freeswitch-esl/blob/main/examples/outbound.rs) in examples folder...
```
pub async fn process_call(conn: freeswitch_esl::EslConnection) {
    log::info!("INCOMING {:?}", conn);
    log::trace!("[{:?}] Start Welcome", std::time::SystemTime::now());
    conn.playback("ivr/ivr-welcome.wav").await.unwrap();
    log::trace!(
        "[{:?}] Complete Welcome. Start sleeping",
        std::time::SystemTime::now()
    );
    tokio::time::sleep(tokio::time::Duration::from_secs(3)).await;
    log::trace!("[{:?}] Complete sleeping", std::time::SystemTime::now());
    conn.playback("ivr/ivr-you_entered.wav").await.unwrap();
    conn.playback("digits/5.wav").await.unwrap();
    log::trace!("[{:?}] PROCESS STOP", std::time::SystemTime::now());
}

#[tokio::main]
async fn main() {
    env_logger::init();
    let addr = "0.0.0.0:8085";
    log::info!("Wait for calls at {:?}", addr);
    let listener = freeswitch_esl::Esl::outbound(addr).await.unwrap();

    loop {
        let (socket, _) = listener.accept().await.unwrap();
        tokio::spawn(async move { nice::process_call(socket).await });
    }
}
```
The FreeSWITCH does play "ivr/ivr-welcome.wav" but my server never reaches "Complete Welcome. Start sleeping" line of code.

**Comment by akmitrich** [comment]
@KaranGauswami seems like I found the root of my problem.
my FreeSWITCH responds with `-Err` on command `event json BACKGROUND_JOB CHANNEL_EXECUTE_COMPLETE` which is issued by [`subscribe` call](https://github.com/KaranGauswami/freeswitch-esl/blob/df6dc403c15e55ea93f03ca4f8b3a579019180e4/src/connection.rs#L173). After that error [channel in `send_recv`](https://github.com/KaranGauswami/freeswitch-esl/blob/df6dc403c15e55ea93f03ca4f8b3a579019180e4/src/connection.rs#L65) awaits forever because [`if let`](https://github.com/KaranGauswami/freeswitch-esl/blob/df6dc403c15e55ea93f03ca4f8b3a579019180e4/src/connection.rs#L100) prevents from sending any command reply on [line 155](https://github.com/KaranGauswami/freeswitch-esl/blob/df6dc403c15e55ea93f03ca4f8b3a579019180e4/src/connection.rs#L155)

**Comment by akmitrich** [comment]
Problem solved: in dialplan there must be `full` argument in socket app args.

**Comment by KaranGauswami** [comment]
Fixed the documentation in #14  to include the additional arguments.
